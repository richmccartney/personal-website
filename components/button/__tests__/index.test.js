import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Button from '../index'

describe('Button:', () => {
  const wrapper = shallow(<Button>Click me!</Button>)

  test('renders snapshot as expected', () => {
    const component = renderer.create(<Button />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders as <button> as expected', () => {
    expect(wrapper.is('button')).toBe(true)
  })

  it('renders with default props as expected', () => {
    expect(wrapper.props().disabled).toEqual(false)
    expect(wrapper.props().type).toEqual('button')
  })
})
