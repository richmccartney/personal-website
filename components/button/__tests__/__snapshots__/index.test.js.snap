// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button: renders snapshot as expected 1`] = `
<button
  disabled={false}
  type="button"
/>
`;
